src/AggregateFunctions/ReservoirSamplerDeterministic.h

```diff
@@ -59,9 +59,10 @@ template <typename T,
     ReservoirSamplerDeterministicOnEmpty OnEmpty = ReservoirSamplerDeterministicOnEmpty::THROW>
 class ReservoirSamplerDeterministic
 {
-    bool good(const UInt32 hash)
+private:
```

Comment:
This is old code (circa 2014), it also contains strange decisions and mistakes. E.g. `total_values` is calculated and serialized but not used.